### PR TITLE
- Add contained support for current  serialization formats.

### PR DESCRIPTION
**Pull request by baldersheim**
- Add support for future compressed formats.

@toregge PR
Extracted from #14181

### CODE REVIEW

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+#pragma once
+
+#include "ichunk.h"
+#include <vespa/vespalib/util/compressionconfig.h>
+
+namespace search::transactionlog {
+
+/// Current default chunk serialisation format
+class XXH64None : public IChunk {
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/chunks.h:11]: Consider adding Chunk suffix to class name.

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+protected:
+    Encoding onEncode(nbostream &os) const override;
+    void onDecode(nbostream &is) override;
+public:
+};
+
+/// TODO Legacy chunk serialisation format to be removed soon.
+class CCITTCRC32None : public IChunk {
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/chunks.h:19]: Consider adding Chunk suffix to class name.

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+protected:
+    Encoding onEncode(nbostream &os) const override;
+    void onDecode(nbostream &is) override;
+public:
+};
+
+/// Future default chunk serialisation format
+class XXH64Compressed : public IChunk {
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/chunks.h:27]: Consider adding Chunk suffix to class name.

File: searchlib/src/vespa/searchlib/transactionlog/ichunk.cpp [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                case Encoding::Compression::none:
+                    return make_unique<XXH64None>();
+                case Encoding::Compression::none_multi:
+                    return make_unique<XXH64Compressed>(CompressionConfig::NONE_MULTI, compressionLevel);
+                case Encoding::Compression::lz4:
+                    return make_unique<XXH64Compressed>(CompressionConfig::LZ4, compressionLevel);
+                case Encoding::Compression::zstd:
+                    return make_unique<XXH64Compressed>(CompressionConfig::ZSTD, compressionLevel);
+                default:
+                    return make_unique<XXH64Compressed>(CompressionConfig::LZ4, compressionLevel);
```
Comment (toregge) [inline searchlib/src/vespa/searchlib/transactionlog/ichunk.cpp:82]: Consider throwing exception instead.

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/chunks.h:11]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/chunks.h:19]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/chunks.h (hunk 3 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/chunks.h:27]: Fixed

File: searchlib/src/vespa/searchlib/transactionlog/ichunk.cpp (hunk 4 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/transactionlog/ichunk.cpp]: Fixed

**Pull request merged**